### PR TITLE
fix(client): get instance wrong when multi instances has the same uri

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/base/uri.py [hunk 1]
```diff
@@ -216,7 +216,11 @@ def real_request_uri(self) -> t.Union[str, Path]:
 
     @property
     def sw_instance_config(self) -> t.Dict[str, t.Any]:
-        return self._sw_config.get_sw_instance_config(self.instance)
+        return (
+            self._sw_config.get_sw_instance_config(self.instance)
+            if not self.instance_alias
```
Comment (jialeicui) [inline client/starwhale/base/uri.py:221]: It seems that the `instance_alias` will always have a value.
https://github.com/star-whale/starwhale/blob/5abf68e650c66ebceed81c18f52d48ee8c6f39ed/client/starwhale/base/uri.py#L106-L109

File: client/starwhale/base/uri.py [hunk 2]
```diff
@@ -216,7 +216,11 @@ def real_request_uri(self) -> t.Union[str, Path]:
 
     @property
     def sw_instance_config(self) -> t.Dict[str, t.Any]:
-        return self._sw_config.get_sw_instance_config(self.instance)
+        return (
+            self._sw_config.get_sw_instance_config(self.instance)
```
Comment (jialeicui) [inline client/starwhale/base/uri.py:220]: How do we define the situation when a user uses the same URL with a different token?
cc @tianweidut

Comment (tianweidut) [reply client/starwhale/base/uri.py:220]: a URL + a token = a instance

Comment (goldenxinxing) [reply client/starwhale/base/uri.py:220]: a URL + a token == an instance alias?

Comment (goldenxinxing) [reply client/starwhale/base/uri.py:220]: if user write uri with the foramt of '${instance URL}/project/...', we can only filter the instance from config by url, but if the format is '${instance-alias}/project/...', We just to use alias to find exactly what the user is referring to。

File: client/starwhale/base/uri.py (hunk 1 above)
Comment (goldenxinxing) [reply client/starwhale/base/uri.py]: done

File: client/starwhale/utils/config.py [hunk 3]
```diff
@@ -147,8 +147,13 @@ def current_instance(self) -> str:
         return str(self._config["current_instance"])
 
     def get_sw_instance_config(self, instance: str) -> t.Dict[str, t.Any]:
-        instance = self._get_instance_alias(instance)
-        return self._config["instances"].get(instance) or {}
+        instance_alias = self._get_instance_alias(instance)
+        return self._config["instances"].get(instance_alias) or {}
```
Comment (jialeicui) [inline client/starwhale/utils/config.py:151]: `return get_sw_instance_config_by_alias(instance_alias)` ?

File: client/starwhale/utils/config.py [hunk 4]
```diff
@@ -147,8 +147,13 @@ def current_instance(self) -> str:
         return str(self._config["current_instance"])
 
     def get_sw_instance_config(self, instance: str) -> t.Dict[str, t.Any]:
-        instance = self._get_instance_alias(instance)
-        return self._config["instances"].get(instance) or {}
+        instance_alias = self._get_instance_alias(instance)
```
Comment (jialeicui) [inline client/starwhale/utils/config.py:150]: `_get_instance_alias` should throw an exception when getting duplicate urls?

Comment (goldenxinxing) [reply client/starwhale/utils/config.py]: done

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**